onnxruntime/core/providers/trt/trt_kernel.cc

```diff
@@ -0,0 +1,259 @@
+                    }
+                }
+                else  //TODO: add other tensor types
+                {
+                    dtype = TensorProto_DataType_UNDEFINED;
+                }
+                tensor_proto.set_name(node.InputDefs()[i]->Name());
+                auto& shape = input_tensor->Shape().GetDims();
+                for (auto& dim : shape)
+                {
+                    tensor_proto.add_dims(dim);
+                }
+                tensor_proto.set_data_type(dtype);
+                graph.AddInitializedTensor(tensor_proto);
+            }
+            else  //graph inputs
+            {
+                graph_input_index_.push_back(i);
+                graph_input_name_.push_back(node.InputDefs()[i]->Name());
+            }
+        }
+    }
+
+    ONNX_NAMESPACE::ModelProto model_p = model.ToProto();
+
+    //Add node's outputs to graphproto's outputs if the node's EdgeEnd nodes are not in the graph
+    std::set<string> output_set;
+    for (int i = 0, end = model_p.graph().output().size(); i < end; ++i)
+    {
+        output_set.insert(model_p.graph().output()[i].name());
+    }
+
+    std::vector<int> output_to_add;
+    for (int i = 0, end = node.OutputDefs().size(); i < end; ++i)
+    {
+        const std::string& output_name = node.OutputDefs()[i]->Name();
+        if (output_set.find(output_name) == output_set.end())
+        {
+            for (int j = 0, end = model_p.graph().value_info().size(); j < end; ++j)
+            {
+                if (output_name == model_p.graph().value_info()[j].name())
+                {
+                    output_to_add.push_back(j);
+                }
+            }
+        }
+    }
+    for (auto& i: output_to_add)
+    {
+        *(model_p.mutable_graph()->mutable_output()->Add()) = model_p.graph().value_info()[i];
+    }
+
+    //Create trt engine
+    string string_buf;
+    model_p.SerializeToString(&string_buf);
+    std::vector<char> onnx_buf(string_buf.begin(), string_buf.end());
+
+    int verbosity = static_cast<int>(nvinfer1::ILogger::Severity::kWARNING);
+    TRTLogger trt_logger(static_cast<nvinfer1::ILogger::Severity>(verbosity));
+    const auto& trt_builder = InferObject(nvinfer1::createInferBuilder(trt_logger));
+    const auto& trt_network = InferObject(trt_builder->createNetwork());
+    const auto& trt_parser  = InferObject(nvonnxparser::createParser(trt_network.get(), trt_logger));
+    trt_parser->parse(onnx_buf.data(), onnx_buf.size());
+    trt_builder->setMaxBatchSize(max_batch_size);
+    trt_builder->setMaxWorkspaceSize(max_workspace_size);
+    engine_ = InferObject(trt_builder->buildCudaEngine(*trt_network.get()));
+    assert(engine_ != nullptr);
+
+    //Build trt context
+    tensorrt_context_ = engine_->createExecutionContext();
+    assert(tensorrt_context_ != nullptr);
+
+    //Get input shape and binding index
+    input_dimension_.resize(trt_network->getNbInputs());
+    for (int i = 0; i < trt_network->getNbInputs(); i++)
```

Comment:
>i = 0; i < trt_network->getNbInputs() [](start = 13, length = 37)

int i = 0, end = trt_network->getNbInputs(); i < end; ++i

The reason you want to do it that way is that it will call getNbInputs on each iteration unless that can be optimized out.

Also good habit in C++ to use prefix increment, so ++i. If 'i' was a class the postfix increment incurs the cost of a copy construct to return the pre-increment value.